### PR TITLE
Feat/data market choice

### LINKED ISSUE #82: Support local collector env flag DATA_MARKET_IN_REQUEST
**Is your feature request related to a problem?**
Local collector should be passed the above flag with value set to `true` in docker compose for multi data market feature to work correctly.

**Describe the solution you'd like**
As described above.

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by Seth-Schmidt**
Fixes #82 #84 

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.8.0 and above.
- [x] I ran pre-commit checks against my changes.
- [x] I've written tests against my changes and all the current present tests are passing.

### Current behaviour
The current setup pattern only supports a single data market and requires manually changing the fields in the .env in order to change to a new data market.

### New expected behaviour
The node will now as which data market should be populated in the .env on start up allowing users to choose by simply selecting the one they want.

#### Changed
- `build.sh` - added prompt to select data market and removed the data market override
- `env.example` - changed the data market and related fields to be dynamic placeholders - added data market in request field
- `docker-compose.yaml` - now passing data market in request env variable to the local collector

## Deployment Instructions
Deployment remains the same besides the inclusion of the additional prompt.

### CODE REVIEW

**Comment by anomit** [comment]
Functionally, this is good to be approved. This has been tested locally as well as on remote instances to be running correctly.

For the PR to be merged though, these need to be finalized:

* The default values being injected by the build script: contract addresses of protocol state, data market etc.
* What about a snapshotter who wishes to start a second snapshotter node from 
  * within the same cloned repo once the `.env` has been generated
  * or a freshly cloned repo where they yet have to be provided a way to run the build script without the local collector

File: env.example
Comment (xadahiya) [inline env.example:13]: If we're overriding all these values in `build.sh` based on user selection, do we need them here?

Comment (anomit) [reply env.example:13]: Yeah I wouldn't say it is necessary to put even the protocol state contract in here. Will review and push changes.

**Comment by Seth-Schmidt** [comment]
> * The default values being injected by the build script: contract addresses of protocol state, data market etc.
> * What about a snapshotter who wishes to start a second snapshotter node from
>   * within the same cloned repo once the `.env` has been generated
>   * or a freshly cloned repo where they yet have to be provided a way to run the build script without the local collector

My initial thought regarding the starting of a second snapshotter from the same cloned repo is to always ask for the data market choice for a single deployment. The multi-script could also be modified to allow the user to input the # of each data market they would like to deploy instead of always alternating between the available markets.

File: env.example
Comment (anomit) [reply env.example:13]: 11f1a7ef addresses this

**Review by anomit — APPROVED** [review]
Tested with multiple combinations of existing pre-mainnet setup envs as well as fresh clones on MacOS and Linux.

Approving. But happy to address any further issues that might come up during review of others.

The following remain to be completed before merge: inject the default protocol state and data market contract address values from `build.sh`.

**Review by xadahiya — APPROVED** [review]
LGTM after the latest changes! Good to go.

**Pull request merged**
